components/gitpod-protocol/src/teams-projects-protocol.ts

```diff
@@ -14,6 +14,7 @@ export interface ProjectConfig {
 }
 
 export interface ProjectSettings {
+    enablePrebuilds?: boolean;
```

Comment:
I feel I don't have :100: of the context, but sharing my thought anyway: I somehow expected this to be `prebuild: PrebuildConfig`:
```
type PrebuildConfig {
  enabled?: boolean;
  ...
}
```

Reply:
@geropl, thanks for the comment. I've created this as WIP, but GitHub extension failed to switch to draft mode. 

Can you elaborate on the need for that? Currently in gitpod-protocol, which is also the base for JSON to be stored in DB, it's a flat list of settings. PAPI introduced another structure for settings, which is not perfectly aligned with server/dashboard, e.g. empty state behaves differently. It seems here already have inconsistency to some degree. Not sure, if that's generally known.  
```
type ProjectSettings struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	Prebuild  *PrebuildSettings  `protobuf:"bytes,1,opt,name=prebuild,proto3" json:"prebuild,omitempty"`
	Workspace *WorkspaceSettings `protobuf:"bytes,2,opt,name=workspace,proto3" json:"workspace,omitempty"`
}
```

What's the benefit of the suggested change to protocol?

And where is the justification for the data migration efforts? Sidebar: I'm trying to reduce the need of data migration to zero if possible in this PR.

Reply:
I was not aware we have pre-existing data to migrate from.
I just felt encapsulating those options feels cleaner.